com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviourUpdater.cs

```diff
@@ -20,38 +25,42 @@ internal void NetworkBehaviourUpdate(NetworkManager networkManager)
             {
                 if (networkManager.IsServer)
                 {
-                    m_Touched.Clear();
-                    for (int i = 0; i < networkManager.ConnectedClientsList.Count; i++)
+                    foreach (var dirtyObj in m_DirtyNetworkObjects)
                     {
-                        var client = networkManager.ConnectedClientsList[i];
-                        var spawnedObjs = networkManager.SpawnManager.SpawnedObjectsList;
-                        m_Touched.UnionWith(spawnedObjs);
-                        foreach (var sobj in spawnedObjs)
+                        for (int i = 0; i < networkManager.ConnectedClientsList.Count; i++)
                         {
-                            if (sobj.IsNetworkVisibleTo(client.ClientId))
+                            var client = networkManager.ConnectedClientsList[i];
+                            if (networkManager.IsHost && client.ClientId == networkManager.LocalClientId)
```

Comment:
Is this safe? There's stuff happening in `PreNetworkVariableWrite()`, but I haven't dug really deep.